app/services/control_rate_service.rb

```diff
@@ -1,25 +1,34 @@
 class ControlRateService
-  CACHE_VERSION = 1
+  CACHE_VERSION = 2
   PERCENTAGE_PRECISION = 1
 
-  # Can be initialized with _either_ a date range or a single date to calculate
-  # control rates. Note that for the date range the returned values will be for each month going back
+  # Can be initialized with _either_ a Period range or a single Period to calculate
+  # control rates. We need to handle a single period for calculating point in time benchmarks.
+  #
+  # Note that for the range the returned values will be for each Period going back
   # to the beginning of registrations for the region.
-  def initialize(region, range: nil, date: nil)
-    raise ArgumentError, "Cannot provide both a range and date" if range && date
-    raise ArgumentError, "Must provide either a range or a single date" if range.nil? && date.nil?
+  def initialize(region, periods:)
     @region = region
     @facilities = region.facilities
-    @range = range
-    @date = date
-    @end_of_date_range = date || range.end
-    logger.info "#{self.class} created for range: #{range} facilities: #{facilities.map(&:id)} #{facilities.map(&:name)}"
+    # Normalize between a single period and range of periods
+    @periods = if !periods.is_a?(Range)
+      @single_period = periods
+      Range.new(periods, periods)
+    else
+      periods
+    end
+    @quarterly_report = @periods.begin.quarter?
+    logger.info "#{self.class} created for periods: #{periods} facilities: #{facilities.map(&:id)} #{facilities.map(&:name)}"
+  end
+
+  def single_period?
```

Comment:
Shouldn't this return a boolean?

Reply:
It returns something that is truthy, which is good for what its used for.